python/lsst/display/firefly/firefly.py

```diff
@@ -105,24 +111,57 @@ def _mtv(self, image, mask=None, wcs=None, title=""):
         """Display an Image and/or Mask on a Firefly display
         """
 
-        if self._regionLayerId:
-            _fireflyClient.removeRegion(self._regionLayerId) # plotId=self.display.frame
-            self._regionLayerId = None
-
-        with tempfile.TemporaryFile() as fd:
-            displayLib.writeFitsImage(fd.fileno(), image, wcs, title)
-            fd.flush()
-            fd.seek(0, 0)
-
-            self._fireflyFitsID = _fireflyClient.uploadFitsData(fd)
-            self._additionalParams=dict(RangeValues=self.__getRangeString(),
-                                        Title=title,
-                                        )
-            ret = _fireflyClient.showFits(self._fireflyFitsID, plotId=self.display.frame,
-                                          additionalParams=self._additionalParams)
-
-        if not ret["success"]:
-            raise RuntimeError("Display failed")
+        if re.search("ImageF", repr(image)):
+            mskimg = afwImage.MaskedImageF(image, mask)
+            exp = afwImage.ExposureF(mskimg, wcs)
+        elif re.search("ImageU", repr(image)):
+            mskimg = afwImage.MaskedImageU(image, mask)
+            exp = afwImage.ExposureF(mskimg, wcs)
+        elif re.search("ImageD", repr(image)):
+            mskimg = afwImage.MaskedImageD(image, mask)
+            exp = afwImage.ExposureF(mskimg, wcs)
+        elif re.search("ImageI", repr(image)):
+            mskimg = afwImage.MaskedImageI(image, mask)
+            exp = afwImage.ExposureF(mskimg, wcs)
+        elif re.search("ImageL", repr(image)):
+            mskimg = afwImage.MaskedImageL(image, mask)
+            exp = afwImage.ExposureF(mskimg, wcs)
+        else:
+            raise RuntimeError("Unknown image type")
+
```

Comment:
You're allocating a lot of memory just to simplify the code (at the worst case, a mask image, a variance image, and misc. small things like a Wcs).  I don't think we want to do this.